README.md

```diff
@@ -43,83 +43,123 @@ tarpc-plugins = { git = "https://github.com/google/tarpc" }
+extern crate tarpc;
+
+use sync_starter::SyncServiceExt;
+use tarpc::server::Options;
 
 fn main() {
-    let (tx, rx) = mpsc::channel();
-    thread::spawn(move || {
-        let mut handle = HelloServer.listen("localhost:0", server::Options::default())
-            .unwrap();
-        tx.send(handle.addr()).unwrap();
-        handle.run();
-    });
-    let client = SyncClient::connect(rx.recv().unwrap(), client::Options::default()).unwrap();
-    println!("{}", client.hello("Mom".to_string()).unwrap());
+    let mut handle = sync_starter::HelloServer.listen("localhost:0", Options::default()).unwrap();
+    println!("Listening on {}", handle.addr());
+    handle.run();
 }
 ```
 
-The `service!` macro expands to a collection of items that form an
-rpc service. In the above example, the macro is called within the
-`hello_service` module. This module will contain `SyncClient`, `AsyncClient`,
-and `FutureClient` types, and `SyncService` and `AsyncService` traits.  There is
-also a `ServiceExt` trait that provides starter `fn`s for services, with an
-umbrella impl for all services.  These generated types make it easy and
-ergonomic to write servers without dealing with sockets or serialization
-directly. Simply implement one of the generated traits, and you're off to the
-races! See the `tarpc_examples` package for more examples.
+In this example we call `listen("localhost:0", ..)` to bind the `HelloServer` object to a port chosen by
+the operating system. We then print the chosen port to stdout so a client can know which port to
+connect to. In a production service it's more common to specify the port as a commandline argument.
+
+Next is the command line client. This client allows you to specify a server address and a person's
+name, and sends a `hello` RPC with that person's name to the specified server. For simplicity this
+binary uses `clap` to handle command line arguments, though it's not required.
+
+```rust
+extern crate clap;
+extern crate tarpc;
+extern crate sync_starter;
+
+use tarpc::client::Options;
+use tarpc::client::sync::ClientExt;
+
+fn main() {
+    let matches = clap::App::new("hello sync client")
+        .arg(clap::Arg::with_name("server_address").required(true))
+        .arg(clap::Arg::with_name("person_name").required(true))
+        .get_matches();
+    let addr = matches.value_of("server_address").unwrap();
+    let person_name = matches.value_of("person_name").unwrap();
+    let mut client = sync_starter::SyncClient::connect(addr, Options::default()).unwrap();
+    println!("{}", client.hello(person_name.into()).unwrap());
+}
+```
+
+Code for this example can be found in `examples/sync_starter/src`. To run it do the following:
+
+```
+$ cd examples/sync_starter
+$ cargo build
+$ ../../target/debug/sync_server
+Listening on [::1]:61081
+```
+
+Now the server is listening on port `61081`. We can then use the generated client to send an RPC to
+the server:
+
+```
+$ ../../target/debug/sync_client "[::1]:61081" Mom
```

Comment:
As above, an alternative is `cargo run --bin sync_client -- "[::1]:61081" Mom`

Also, that port probably won't work for them. This walkthrough should probably have them specify the port arg when starting the server so that this line will work without them having to do anything other than copy paste.